src/main.rs

```diff
@@ -549,11 +550,21 @@ fn win_in_n<const S: usize>(
     let mut legal_moves = vec![];
     let mut tinue_moves = vec![];
 
-    position.generate_moves(&mut legal_moves);
+    let mut moves_with_heuristic_scores = vec![];
+
+    position.generate_moves_with_probabilities(
+        &position.group_data(),
+        &mut legal_moves,
+        &mut moves_with_heuristic_scores,
+    );
+
+    // Sort the moves using Tiltak's heuristic
+    // Checking the best moves first gives a ~35% speedup for depth 5
+    moves_with_heuristic_scores.sort_by_key(|(_mv, score)| 0 - (score * 1024.0) as i64);
```

Comment:
1. Why do you cast it to `i64`? Are integer comparisons faster than floats? Otherwise `-score` would probably be sufficient.

2. [Documentation](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.sort_by_key) recommends using `sort_unstable_by_key` for a smaller CPU/memory footprint. I don't think stable-ness is required here, so we should probably use it.

Reply:
It's a terrible, ugly hack to make up for the fact that floats can't be sorted easily in Rust, because they don't implement [cmp::Ord](https://doc.rust-lang.org/stable/std/cmp/trait.Ord.html). Ints can, however, so I just cast it. I'll change it to make it more clear.

Unstable sort is just an improvement, thanks!

Reply:
That's good to know!